### DISCUSSION TITLE
Undici request - use another public IP address

### DISCUSSION THREAD

**Discussion by qualitydude** [nodejs/undici#1577] [Q&A]
Hi,

I was trying to use undici request with option to select which IP address should be used to request but without any success.

Example how I am doing it with AXIOS:
`	
await axios(url, {
	httpsAgent: new https.Agent({ localAddress: "XXX.XXX.XXX.XXX", keepAlive: true, keepAliveMsecs: 30000 })
})
	.then(res => {})
`
 any possibilities of that with undici?

**Comment by keverw** [comment]
Did you ever figure this out? I am wanting to do the same for a project. My idea was to just use the IP in the hostname, replacing the domain and set the host header to the domain, but I figured that would be a bit hacky and was worried how it'd work on https:// domains, etc.

**Comment by metcoder95** [comment]
`localAddress` is already supported by undici since `v5.12` through `Client`
